digdag-server/src/main/java/io/digdag/server/Authenticator.java

```diff
@@ -1,51 +1,62 @@
 package io.digdag.server;
 
-import java.util.Objects;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import io.digdag.client.config.Config;
+import org.immutables.value.Value;
+
+import javax.annotation.Nullable;
 import javax.ws.rs.container.ContainerRequestContext;
 
+import static org.immutables.value.Value.Style.ImplementationVisibility.PACKAGE;
+
 public interface Authenticator
 {
-    public static class Result
+    @Value.Immutable
+    @Value.Style(visibility = PACKAGE)
```

Comment:
This interface needs to be public so that a custom authenticator module can instantiate it.